lib/util/state.js

```diff
@@ -585,6 +586,24 @@ const getAgencyIdsForFilter = (name, transitOperators) => {
     .map((x) => x.agencyId)
 }
 
+/**
+ * Determine if a string matches a query. Uses direct matching if the string is only numbers,
+ * uses fuzzy matching otherwise.
+ */
+const isFilterMatch = (string, query) => {
+  const normalizedQuery = query.toLowerCase()
+  const normalizedString = string.toLowerCase()
```

Comment:
I'm wondering could we also filter dashes and spaces out of the search query + results? Just looking at the "1 line" search I know this PR inspired, and the results don't seem quite as intuitive as we'd hope. Removing spaces and punctuation could also allow for upping the similarity ratio from .4 to something a little higher? 
![image](https://user-images.githubusercontent.com/115499534/229860576-0bdecb88-50d5-47d6-9482-79c3f7895e76.png)

Reply:
Also so glad we talked about the correlation between hovering and these filters today, I thought I was going insane while reviewing this.

Reply:
Sure sure, to be clear I think this works great as-is! But right now for this search, every result has a match of `.75`. If you remove spaces and dashes, `1-Line` has a match of `1`, but `T-Line` would still be `.75`

Reply:
Although I guess then you'd have to sort the results based on this percentage

Reply:
Yeah this issue is part of the reason why I was against doing fuzzy search way back when this was originally built. I think a text similar algorithm that could be tweaked for each set of routes might get the results that we're looking for, but yet another instance-specific method like that doesn't make me happy. I think our solution needs to be general